src/app/qgssettingstree.cpp

```diff
@@ -166,6 +173,63 @@ void QgsSettingsTree::updateSetting( QTreeWidgetItem *item )
     refresh();
 }
 
+void QgsSettingsTree::showContextMenu( QPoint pos )
+{
+  QTreeWidgetItem *item = itemAt( pos );
+  if ( !item )
+    return;
+
+  Type itemType = static_cast< Type >( item->data( 0, TypeRole ).toInt() );
```

Comment:
You can use item->data( 0, TypeRole ).value<Type>()

Reply:
thanks!